src/eventmodel.cpp

```diff
@@ -80,6 +80,7 @@ QHash<int, QByteArray> EventModel::roleNames() const
     roles[BaseRole + FromVCardLabel] = "fromVCardLabel";
     roles[BaseRole + ReadStatus] = "readStatus";
     roles[BaseRole + SubscriberIdentity] = "subscriberIdentity";
+    roles[BaseRole + ContactName] = "contactName";
```

Comment:
Ehm, speaking of that role vs column, what the heck has been happening here. From the git history looks like initial commit was having columns and then later also roles got added. Makes me wonder if the initial thing was done wrong and then never properly fixed.

Reply:
@pvuorela  As far as I understand, the Tree model seems to assume here, if there are columns, then it should display them as columns.

Reply:
And if only the lines, then only they will be displayed, but my opinion is the solution is very strange, it looks sloppy.

Reply:
I'd like to get that sorted out, but guess it shouldn't be blocking adding new things.

However for the new thing is there a reason why it's a column and not a role?

Reply:
so that you can get the value from outside ```sourceModel()->index(sourceRow, 0).data(roleId).toString()```

Reply:
Yea but now it's not a roleId, it's a ColumnId. Above might work but really by the api it should be something like index(sourceRow, ContactName).data(/*any integer below BaseRole to work */), right?

Reply:
Yes that's right

Reply:
And still to add, your example above would work more by the book if this new thing was a role instead?

Reply:
At the moment, it is more convenient to get the value from one enum, since, for example, from it, for example, RemoteUid is used

Reply:
Not sure I'm following. So what I'm mostly suggesting is moving the new role name to the latter enum, after DateAndAccountGroupingRole, and then declaring the usage here without any BaseRole extra addition. That is, like the roles are commonly used, avoiding the column strangeness.

Reply:
Done